### PR TITLE
Implement nitrogen reduction for grassland renewal and destruction

### LINKED ISSUE #412: Implement Nitrogen Discount (Korting) for Grassland Renewal/Destruction (2025 & 2026)
## Context

Dutch nitrogen regulations apply a reduction ("korting") to the **stikstofgebruiksnorm** (nitrogen usage standard) when grassland is destroyed (scheuren). This is because the decomposing sod releases significant nitrogen (mineralization) for the next crop.

* **Renewing (Gras-na-Gras):** Old grass -> New grass. Discount: **50 kg N/ha**.
* **Destruction (Gras-naar-Bouwland):** Old grass -> Maize/Potato. Discount: **65 kg N/ha**.

We need to implement this logic in `fdm-calculator` for both 2025 and 2026.
**Crucially**, the rules differ between 2025 and 2026 due to the end of derogation.

---

## 1. Logic for 2025

**File:** `fdm-calculator/src/norms/nl/2025/value/stikstofgebruiksnorm.ts`
**Function:** `calculateKorting`

Update the function to include the following logic alongside the existing "vanggewas" check.

### A. Grassland Renewal (50 kg N/ha)

* **Condition:** Old crop was Grass AND New crop is Grass ("Direct herinzaai").
* **Period:** Old grass `b_lu_end` is between **June 1 and August 31** (check specific dates per soil type below).
* **Soil Type Logic:**
  * **Sand / Loess:** Always applies.
  * **Clay / Peat:**
    * **Derogation Farm + NV-Area:** June 1 – August 31.
    * **Derogation Farm + Non-NV-Area:** June 1 – September 15.
    * **Non-Derogation Farm:** February 1 – September 15.
  * *Note:* We have `is_derogatie_bedrijf` and `is_nv_area` available in the input, so please implement these exact date ranges.

### B. Grassland Destruction (65 kg N/ha)

* **Condition:** Old crop was Grass AND New crop is Maize OR **Specific Potato Types**.
* **Potato Restriction:** The discount applies **ONLY** to:
  * Consumption Potatoes (Consumptieaardappelen)
  * Factory Potatoes (Fabrieksaardappelen)
  * **EXCLUDED:** Seed Potatoes (Pootaardappelen) do **NOT** trigger this discount (Article 28f lid 1a explicitly omits them).
* **Soil Type Logic:**
  * **Sand / Loess:** Applies if `b_lu_end` is between **Feb 1 and May 10**.
  * **Clay / Peat:** Applies if `b_lu_end` is between **Feb 1 and May 31** (Non-NV) or **Mar 15** (NV - check specific year rules).

---

## 2. Logic for 2026

**File:** `fdm-calculator/src/norms/nl/2026/value/stikstofgebruiksnorm.ts`
**Function:** `calculateKorting`

**Major Change:** In 2026, Article 28f restricts these discounts **exclusively to Sand and Loess soils**. Clay and Peat are excluded.

### A. Grassland Renewal (50 kg N/ha)

* **Soil Check:** `region` MUST be `zand_nwc`, `zand_zuid`, or `loess`. (Return 0 discount if Clay/Peat).
* **Crop Sequence:**
  * `PreviousCultivation`: Grass (`b_lu_catalogue` in grass list).
  * `CurrentCultivation`: Grass.
  * `Gap`: `Current.b_lu_start` is strictly after `Previous.b_lu_end`.
* **Timing:** `Previous.b_lu_end` is between **June 1 and August 31**.
* **Discount:** **50 kg N/ha**.

### B. Grassland Destruction (65 kg N/ha)

* **Soil Check:** `region` MUST be `zand_nwc`, `zand_zuid`, or `loess`.
* **Crop Sequence:**
  * `PreviousCultivation`: Grass.
  * `CurrentCultivation`: Maize (`mais`) OR **Eligible Potato** (Consumptie/Fabriek).
* **Potato Restriction:** Verify `CurrentCultivation` is NOT "Pootaardappelen". Use `b_lu_catalogue` or variety type from `nitrogenStandardsData`.
* **Timing:** `Previous.b_lu_end` must be early in the year (typically **Feb 1 - May 10** for Sand).
* **Exclusion (Catch Crop):**
  * Check if `PreviousCultivation` was a "Vanggewas" (catch crop) or "Groenbemester" sown the previous autumn.
  * *Implementation:* Check `PreviousCultivation.is_vanggewas` (if available in standard data) or infer from sowing date (late previous year). If it was a catch crop, **NO discount** applies (Article 28f lid 2).
* **Discount:** **65 kg N/ha**.

## 3. Error Handling (Invalid Dates)

If the user performs Grassland Renewal or Destruction **outside** the legally allowed periods (e.g., renewing grass in December), the calculator should **throw a descriptive error**.

* **Reason:** These actions are legally restricted. Calculating a norm without warning might mislead the user into thinking their plan is compliant.
* **Implementation:**
  * Detect the "Renewal" or "Destruction" event (as per logic above).
  * Check `Previous.b_lu_end` against the valid date range for the specific Soil Type and Year.
  * **If Invalid:** Throw an error (e.g., `"Grassland renewal on Sand is only allowed between June 1 and August 31."`).

---

## Implementation Details for Developers

1. **Identify Grassland:** Use `nitrogenStandardsData` to check if a `b_lu_catalogue` belongs to type "grasland".
2. **Identify Potatoes:**
    * Check `nitrogenStandardsData` for potato types.
    * Filter OUT `Pootaardappelen` (Seed Potatoes) for the destruction discount.
3. **Sequence Detection:**
    * Sort `cultivations` by `b_lu_start`.
    * Find the transition where one crop ends (`b_lu_end`) and the next begins.
    * The "Destruction Date" is the `b_lu_end` of the first crop.
4. **Soil Mapping:**
    * `sandyOrLoessRegions = ["zand_nwc", "zand_zuid", "loess"]`
    * `clayOrPeatRegions = ["klei", "veen"]`
5. **Mutually Exclusive Logic:**
    * The rules are structurally mutually exclusive based on the **Current Cultivation** (the crop *after* the grass):
        * Next is **Grass**: Apply Renewal (50kg).
        * Next is **Maize/Eligible Potato**: Apply Destruction (65kg).

## Acceptance Criteria

* [ ] 2025 Calculator applies 50kg/65kg discount on Sand/Loess AND Clay/Peat (under derogation conditions).

* [ ] 2026 Calculator applies 50kg/65kg discount **ONLY** on Sand/Loess.
* [ ] 2026 Calculator does **NOT** apply 65kg discount if the destroyed grass was a catch crop.
* [ ] **Potato Check:** Calculator applies 65kg discount for Consumption/Factory potatoes but **NOT** for Seed Potatoes (Pootaardappelen).
* [ ] Unit tests added to `stikstofgebruiksnorm.test.ts` for both years covering these scenarios.

### PR DESCRIPTION
**Pull request by SvenVw**
## Summary by CodeRabbit

* **New Features**
  * Implemented grassland renewal (50 kg N/ha) and destruction (65 kg N/ha) reductions with region- and year-specific date windows, NV-area and derogation rules; 2026 limited to sandy/loess regions and includes expanded catch‑crop/sowing-date handling.

* **Documentation**
  * Updated nitrogen usage norms to describe renewal/destruction reductions, soil-type windows, derogation/NV considerations, and localized Dutch error messages for invalid dates.

* **Tests**
  * Added extensive 2025/2026 test coverage across regions, dates, crops, and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

Closes #412

### CODE REVIEW

File: fdm-calculator/src/norms/nl/2025/value/stikstofgebruiksnorm.ts [hunk 1]
```diff
@@ -443,11 +443,176 @@ function determineSubTypeOmschrijving(
...
+            if (isValidRenewal) {
+                return {
+                    amount: new Decimal(50),
+                    description: ". Korting: 50kg N/ha: graslandvernieuwing",
+                }
+            }
+        }
+
+        // 2. Grassland Destruction (Gras-naar-Bouwland) -> 65 kg N/ha korting
+        // Applies if New Crop is Maize OR Consumption/Factory/Starch Potatoes (NOT Seed Potatoes)
```
Comment (gerardhros) [inline fdm-calculator/src/norms/nl/2025/value/stikstofgebruiksnorm.ts]: check question: is this korting also applicable for fertilizer recommendation? usually the current endpoint ignores the past if i remember it well
